File: client/lua/krpc/test/test_connection_tcpip.lua
local class = require 'pl.class'
local Connection = require 'krpc.connection'
local connectiontest = require 'krpc.test.connectiontest'

-- The connection carried over TCP/IP, against a server the test listens on itself rather
-- than a kRPC server
local TestConnectionTCPIP = class(connectiontest.ConnectionTest)

function TestConnectionTCPIP:start_server()
  local listener, address, port = connectiontest.tcpip_listener()
  self._address = address
  self._port = port
  return connectiontest.EchoServer(listener)
end

function TestConnectionTCPIP:stop_server()
end

function TestConnectionTCPIP:connect()
  local conn = Connection(self._address, self._port)
  conn:connect()
  return conn
end

return TestConnectionTCPIP

File: client/python/krpc/test/test_connection_tcpip.py
import socket
import unittest
from krpc.connection import Connection
from krpc.test.connectiontest import ConnectionTest, EchoServer


class TestConnectionTCPIP(ConnectionTest):
    """The connection carried over TCP/IP, against a server the test listens on itself
    rather than a kRPC server."""

    __test__ = True

    @classmethod
    def setUpClass(cls) -> None:
        cls.server = EchoServer(socket.AF_INET)
        cls.server.start()

    @classmethod
    def tearDownClass(cls) -> None:
        cls.server.stop()

    def connect(self) -> Connection:
        address, port = self.server.address
        conn = Connection(address, port)
        conn.connect()
        return conn


if __name__ == "__main__":
    unittest.main()
